src/components/ComposerAttachments.vue

```diff
@@ -93,22 +99,58 @@ export default {
 		onAddLocalAttachment() {
 			this.$refs.localAttachments.click()
 		},
-		fileNameToAttachment(name, id) {
+		fileNameToAttachment(name, size, id) {
 			return {
 				fileName: name,
 				displayName: trimStart('/')(name),
 				id,
+				size,
 				isLocal: id !== undefined,
 			}
 		},
 		emitNewAttachment(attachment) {
 			this.$emit('input', this.value.concat([attachment]))
 		},
+		totalSizeOfUpload() {
+			return Object.values(this.value).reduce((acc, upload) => {
+				if (!upload.isLocal) {
+					// Ignore link shares
+					return acc
+				}
+
+				return acc + upload.size
+			}, 0)
+		},
 		onLocalAttachmentSelected(e) {
 			this.uploading = true
 
 			Vue.set(this, 'uploads', {})
 
+			const toUpload = Object.values(e.target.files).reduce((acc, file) => {
+				return acc + file.size
+			}, 0)
+			const newTotal = toUpload + this.totalSizeOfUpload()
+			logger.debug('checking upload size limit', {
```

Comment:
@GretaD what does this line log for you?

Reply:
[DEBUG] mail: checking upload size limit 
Object { app: "mail", uid: "greta", existingUploads: 0, toUpload: 20436, limit: "0", newTotal: 20436 }

Reply:
also, sorry i forgot to include in the first comment, when i click New Message is shows this: 
```
[Vue warn]: Invalid prop: type check failed for prop "uploadSizeLimit". Expected Number with value 0, got String with value "0".

found in

---> <ComposerAttachments> at src/components/ComposerAttachments.vue
```

Reply:
We need to cast. On it.

Reply:
Please try again.